dataclients/kubernetes/definitions/routegroups.go

```diff
@@ -463,15 +464,7 @@ func ValidateRouteGroups(rl *RouteGroupList) error {
 	var err error
 	// avoid the user having to repeatedly validate to discover all errors
 	for _, i := range rl.Items {
-		nerr := ValidateRouteGroup(i)
-		if nerr != nil {
-			err = errors.Wrap(err, nerr.Error())
-		}
-	}
-
-	if err != nil {
-		return err
+		err = errorsJoin(err, ValidateRouteGroup(i))
 	}
-
-	return nil
+	return err
```

Comment:
This should be like
```go
var errs []error
for _, i := range rl.Items {
    errs = append(errs, ValidateRouteGroup(i))
}
return errorsJoin(errs...)
```

Reply:
both should work actually.

Reply:
Yes, but then we decide to change `errors.New(strings.Join(errVals, "\n"))` to `fmt.Errorf("multiple errors: %s", strings.Join(errVals, "\n"))` and message becomes screwed up.
The helper as well as errors.Join is designed to handle multiple errors, not two so lets use it.